jerry-core/ecma/builtin-objects/ecma-builtin-helpers.c

```diff
@@ -416,45 +416,68 @@ ecma_builtin_helper_array_concat_value (ecma_object_t *array_obj_p, /**< array *
                                         ecma_value_t value) /**< value to concat */
 {
   /* 5.b */
-  if (ecma_is_value_object (value))
+#if ENABLED (JERRY_ES2015_BUILTIN)
+  ecma_value_t is_spreadable = ecma_object_is_concat_spreadable (value);
+
+  if (ECMA_IS_VALUE_ERROR (is_spreadable))
   {
-    ecma_object_t *obj_p = ecma_get_object_from_value (value);
+    return is_spreadable;
+  }
 
-    if (ecma_get_object_type (obj_p) == ECMA_OBJECT_TYPE_ARRAY)
-    {
-      /* 5.b.ii */
-      uint32_t arg_len = ecma_array_get_length (obj_p);
+  bool spread_object = is_spreadable == ECMA_VALUE_TRUE ? true : false;
 
-      /* 5.b.iii */
-      for (uint32_t array_index = 0; array_index < arg_len; array_index++)
-      {
-        /* 5.b.iii.2 */
-        ecma_value_t get_value = ecma_op_object_find_by_uint32_index (obj_p, array_index);
+  ecma_free_value (is_spreadable);
```

Comment:
if the `is_spreadable` is not error the result must be `ECMA_VALUE_{TRUE_FALSE}`. Those are simple values so no need to free them.